README.md

```diff
@@ -98,42 +88,341 @@ See [variables.tf] and [examples/] for details and use-cases.
+  - "intra" defines a set of subnets that have no connectivity to the public Internet.
+
+  Default is `"private"`.
+
+- **`map_public_ip_on_launch`**: *(Optional `bool`)*
+
+  Whether resources deployed into the subnet will be assigned a public IPv4 address when launched.
+  Default is `true` when subnet class is `public`, `false` otherwise.
+
+- **`cidr_block`**: *(Optional `string`)*
+
+  Define the base CIDR Block of the subnets and the parameters to calculate each CIDR Block.
+  Default is the CIDR Block of the VPC (`cidr_block`).
+
+- **`newbits`**: *(Optional `number`)*
+
+  How many bits should be added when calculating the subnets CIDR Blocks.
+  Default is `8`.
+
+- **`netnums_by_az`**: **(Required `map(list(number)`)**
+
+  A map of subnets keyed by availability zone suffix (a,b,c,d,e,f).
+  The numbers define the network number with in the CIDR Block.
+  See https://www.terraform.io/docs/configuration/functions/cidrsubnet.html for details on how this is calculated internally
+
+  Note: When adjusting cidr_block or newbits you might also need to adjust the netnums.
+  The example shows how to deploy one subnet in availability zone "a" ("10.0.0.0/24") and one subnet in availability zone "b" ("10.0.1.0/24")
+
+  ```
+  cidr_block = "10.0.0.0/16"
+  newbits    = 8
+
+  netnums_by_az = {
+    a = [0] # "10.0.0.0/24"
+    b = [1] # "10.0.1.0/24"
+  }
+  ```
+
+- **`tags`**: *(Optional `map(string`)*
+
+  A map of tags that will be applied to each subnet in this group-class combination.
+  Those tags will be merged with a `Name` tag, `module_tags`, `subnet_tags` and tags for the subnet class `public_subnet_tags`, `private_subnet_tags`, or `intra_subnet_tags`.
+  Default is `{}`.
+
+##### Internet Gateway
+
+- **`internet_gateway_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created Internet Gateway.
+  An Internet Gateway is created if a public subnet is defined.
+  All public egress and ingress traffic of all public subnets will be routed through the same Internet Gateway.
+  Default is `{}`.
+
+##### NAT Gateways
+
+- **`nat_gateway_mode`**: *(Optional `string`)*
+
+  Set the mode for the NAT Gateways. NAT Gateways will only be created when private subnets are defined.
+  Each private subnet needs at least one configured public subnet in the same availability zone.
+  Each NAT gateway will be assigned an Elastic IP Address.
+  When changing the mode, NAT Gateways and Elastic IP Addresses will be created or destroyed.
+  All public egress traffic of all private subnets will be routed through the same set of NAT Gateways.
+  Possible modes are `none`, `single`, or `one_per_az`. Choose
+
+  - `none` to create no NAT Gateways at all (use for debugging only),
+  - `single` to create a single NAT Gateway inside the first defined Public Subnet, or
+  - `one_per_az` to create one NAT Gateway inside the first Public Subnet in each Availability Zone.
```

Comment:
should we make it clear that additional nat gateways produce significant cost?

Reply:
good point.. can add this to the todo list..